ui/js/reactrevisions.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+// via pushlogRevisionsClone.html
+var pushlogRevisionComponent = React.createClass({
+    displayName: 'pushlogRevisionComponent',
+    propTypes: {
+        href: React.PropTypes.string.isRequired
+    },
+    render: function() {
+        return React.DOM.li(
+            null,
+            React.DOM.a({
+                href: this.props.href,
+                'data-ignore-job-clear-on-click': true,
+                target: '_blank'
+            },
+            '\u2026and more',
+            React.DOM.i({ className: 'fa fa-external-link-square' })
+            )
+        );
+    }
+});
+var pushlogRevisionComponentFactory = React.createFactory(pushlogRevisionComponent);
+
+// via revisionsClone.html
+var revisionItemComponent = React.createClass({
+    displayName: 'revisionItemComponent',
+    propTypes: {
+        revision: React.PropTypes.object.isRequired,
+        repo: React.PropTypes.object.isRequired,
+        linkifyBugsFilter: React.PropTypes.func.isRequired,
+        initialsFilter: React.PropTypes.func.isRequired
+    },
+    render: function() {
+        var email, name, userTokens, escapedComment, escapedCommentHTML, initialsHTML, tags;
+
+        userTokens = this.props.revision.author.split(/[<>]+/);
+        name = userTokens[0];
+        if (userTokens.length > 1) email = userTokens[1];
+        initialsHTML = { __html: this.props.initialsFilter(name) };
+
+        escapedComment = _.escape(this.props.revision.comments.split('\n')[0]);
+        escapedCommentHTML = { __html: this.props.linkifyBugsFilter(escapedComment) };
+
+        tags = "";
+        if (escapedComment.search("Backed out") >= 0 ||
+            escapedComment.search("Back out") >= 0) {
+            tags += "backout ";
+        }
+        tags = tags.trim();
+
+        return React.DOM.li(
```

Comment:
Well, it will be a little tricky to get JSX to compile in our current ``grunt`` model.  I had tried and hit several speed bumps.  Enough that I didn't think it was worth the overhead.  However, if we write more and more and it DOES seems worth it, we can address it at that time.

If this seems work-able, let's stick with straight JS and React.DOM work.  Saves us some infrastructure work now.  :)